Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -3067,12 +3067,10 @@ static void dataDetectorImageOverlayPositionInformation(const HTMLElement& overl
 
     auto [foundElement, elementBounds] = *elementAndBounds;
     auto identifierValue = parseInteger<uint64_t>(foundElement->attributeWithoutSynchronization(HTMLNames::x_apple_data_detectors_resultAttr));
-    if (!identifierValue)
+    if (!identifierValue || !*identifierValue)
         return;
 
-    auto identifier = LegacyNullableObjectIdentifier<ImageOverlayDataDetectionResultIdentifierType>(*identifierValue);
-    if (!identifier.isValid())
```

Comment:
Is the `isValid()` check not needed when using `ObjectIdentifier`?

Reply:
ObjectIdentifier doesn't have an "invalid" state anymore. That's the difference with LegacyNullableObjectIdentifier. You cannot default-construct a ObjectIdentifier or construct it with 0.

Reply:
Note that isValid() was checking that the internal integer wasn't 0. Now I added a `|| !*identifierValue` check above to make sure we can't pass 0 to the ObjectIdentifier constructor.